NFT List - Fix when too many images are displayed

redDwarf03 opened the pull request:
# Description

Fixes #752

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update
- Refactorization

## Material used:

Please delete options that are not relevant.
- iOS (Smartphone/Tablet)
- Android (Smartphone/Tablet)
- Browser
- Linux
- Windows
- iOS (Mac)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Test A
- Test B

### Patrol

Please list here the tests created in Patrol for this pull request.

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

## Useful info for the reviewer:

redDwarf03 committed:
Internalize dynamic_height_grid_view widget

redDwarf03 committed:
Internalize dynamic_height_grid_view widget

redDwarf03 committed:
Add cache management

redDwarf03 committed:
Add cache management

redDwarf03 committed:
Loading NFT - Optimize performance

redDwarf03 committed:
Loading NFT - Optimize performance

redDwarf03 committed:
Riverpod generator 2.2.4 Disable all lints in generated files.

redDwarf03 committed:
Riverpod generator 2.2.4 Disable all lints in generated files.

redDwarf03 committed:
Display properties

redDwarf03 committed:
Display properties

redDwarf03 committed:
Improve UI/UX

redDwarf03 committed:
Improve UI/UX

redDwarf03 committed:
fix order

redDwarf03 committed:
fix order

redDwarf03 committed:
update depencies

redDwarf03 committed:
update depencies

redDwarf03 committed:
Id token management

redDwarf03 committed:
Id token management

redDwarf03 committed:
clean dependencies

redDwarf03 committed:
clean dependencies

redDwarf03 committed:
move files

redDwarf03 committed:
move files

redDwarf03 committed:
generate pubspec

redDwarf03 committed:
generate pubspec

redDwarf03 committed:
fix due rebase

redDwarf03 committed:
fix due rebase

redDwarf03 committed:
Fix properties name (due to riverpod changes)

redDwarf03 committed:
Fix properties name (due to riverpod changes)

redDwarf03 committed:
rebase

redDwarf03 committed:
rebase

redDwarf03 committed:
remove Iconsax

redDwarf03 committed:
remove Iconsax

redDwarf03 committed:
lint

redDwarf03 committed:
lint

redDwarf03 committed:
fix null exception

redDwarf03 committed:
fix null exception

redDwarf03 committed:
Fix NFT and collection loading

redDwarf03 committed:
Fix NFT and collection loading

gdaguin commented on lib/application/nft/nft.dart:
```diff
@@ -20,17 +27,190 @@ Future<TokenInformations?> _getNFT(
+}
+
 class NFTRepository {
   Future<TokenInformations?> getNFT(
```
As the next method is called getNFTList, I was waiting for this method to return singles AccountToken item. Instead it is returning a different kind of object.
I think that this will be a good idea to rename this method getNFTInfo to not being confused.

gdaguin committed:
Massive code simplification

gdaguin committed:
Massive code simplification

gdaguin committed:
Code refacto

+ Renaming getNFT function into getNFTInfo
+ Removing a lot of warnings in the code (by just renaming informations into information)

gdaguin committed:
Code refacto

+ Renaming getNFT function into getNFTInfo
+ Removing a lot of warnings in the code (by just renaming informations into information)

gdaguin commented:
@redDwarf03, all the code is ok for me.

gdaguin committed:
Merge branch '752-nft-list-display-problem' of https://github.com/archethic-foundation/archethic-wallet into 752-nft-list-display-problem

gdaguin committed:
Bug fix

- Display only one collection when we have a collection of NFT, instead of x collections containing x NFT
- Bug fix after code simplification

gdaguin committed:
Bug fix on nftInfoOffChainList

gdaguin committed:
Thumbnail for NFT collection

gdaguin committed:
Fix a bug when a contact has special characters in his name

When the user taps on a contact detail, we were not able to find the contact

redDwarf03 commented on lib/model/data/appdb.dart:
```diff
@@ -184,7 +185,9 @@ class DBHelper {
-      if (contact.name.toLowerCase() == nameWithAt.toLowerCase()) {
+      // decode : hive encode stored strings, we need to decode it
+      if (contact.name.decode().toLowerCase() ==
+          nameWithAt.decode().toLowerCase()) {
```
Normally nameWithAt is not encoded and we have always ContactFormatters to decode contact name

redDwarf03 commented on lib/util/string_util.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/foundation.dart';
+
+extension StringExt on String {
+  String decode() {
```
Decode could be used by a lot of decoding algorithms. So i think this method is too specific for a generic concept.
We can use existing class `ContactFormatters`

redDwarf03 committed:
Lint

redDwarf03 committed:
https://github.com/archethic-foundation/archethic-wallet/issues/786

redDwarf03 committed:
Replace String extension with existing Contact extension

redDwarf03 committed:
Fix bugs

gdaguin committed:
Fix a bug where a transfered NFT from a collection where still visible for the giver

Fix also (and again) a bug with account changes

gdaguin committed:
Fix a bug when tapping on a searched NFT

gdaguin committed:
Fix the name of the NFT that is going to be sent from the transfer confirm sheet

redDwarf03 committed:
lint
